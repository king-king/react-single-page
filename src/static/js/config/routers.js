import asyncComponent from 'utils/common/asyncComponent';


export default {
    '/wq/page1': {
        RenderComponent: asyncComponent(() => import('../template/index/components/Page1'))
    },
    '/wq/page2': {
        RenderComponent: asyncComponent(() => import('../template/index/components/Page2'))
    }
};
