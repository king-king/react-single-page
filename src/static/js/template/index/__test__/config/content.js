export default {};

// 初始化（无数据的情况）
export const initConfig = {
};


// 获取到数据后（都有数据的情况）
export const fullConfig = {
};
