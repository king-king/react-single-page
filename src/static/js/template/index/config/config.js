/*
 * Created by wangqun  At Tue Feb 18 2020 16:20:56
 */

export const url = {};

export const defaultData = {};
