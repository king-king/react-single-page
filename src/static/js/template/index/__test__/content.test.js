import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.2';
import Content from '../components/Content';
import { initConfig, fullConfig } from './config/content';

Enzyme.configure({ adapter: new Adapter() });

describe('Content test', () => {
    test('快照', () => {
        const wrapper = mount(<Content {...initConfig} />);
        expect(wrapper.render()).toMatchSnapshot();
        wrapper.setProps(fullConfig);
        expect(wrapper.render()).toMatchSnapshot();
    });

    test('交互', () => {
    });

    test('生命周期', () => {
    });
});
