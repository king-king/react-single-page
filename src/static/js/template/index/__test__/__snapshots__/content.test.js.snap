// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content test 快照 1`] = `
<div
  class="section"
>
  <div
    class="section-title"
  >
    TITLE
  </div>
</div>
`;

exports[`Content test 快照 2`] = `
<div
  class="section"
>
  <div
    class="section-title"
  >
    TITLE
  </div>
</div>
`;
